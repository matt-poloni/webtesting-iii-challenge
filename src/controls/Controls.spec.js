import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import Controls from './Controls';
import Dashboard from '../Dashboard/Dashboard';

describe('<Controls />', () => {
  it('should render', () => {
    render(<Controls />);
  })

  describe('Close/Open Gate button', () => {
    it("should provide a button to toggle the 'closed' state", () => {
      const { getByText } = render(<Controls />);
      const button = getByText(/Close Gate/);
      expect(button.nodeName).toBe('BUTTON');
    })

    it("should be disabled if passed a 'true' locked prop", () => {
      const { getByText } = render(<Controls locked={true} />);
      const closeBtn = getByText(/Close Gate/);
      const lockBtn = getByText(/Unlock Gate/);
      expect(closeBtn.disabled).toBe(true);
    })
    
    it("should toggle the button's text when clicked", () => {
      const { getByText } = render(<Dashboard />);
      const button = getByText(/Close Gate/);
      fireEvent.click(button);
      getByText(/Open Gate/);
      fireEvent.click(button);
      getByText(/Close Gate/);
    })
  })

  describe('Lock/Unlock Gate button', () => {
    it("should provide a button to toggle the 'locked' state", () => {
      const { getByText } = render(<Controls />);
      const button = getByText(/Lock Gate/);
      expect(button.nodeName).toBe('BUTTON');
    })
    
    it("should be disabled if passed a 'false' closed prop", () => {
      const { getByText } = render(<Controls closed={false} />);
      const closeBtn = getByText(/Close Gate/);
      const lockBtn = getByText(/Lock Gate/);
      expect(lockBtn.disabled).toBe(true);
    })

    it("should toggle the button's text when clicked", () => {
      const { getByText } = render(<Dashboard />);
      const closeBtn = getByText(/Close Gate/);
      const lockBtn = getByText(/Lock Gate/);
      fireEvent.click(closeBtn);
      fireEvent.click(lockBtn);
      getByText(/Unlock Gate/);
      fireEvent.click(lockBtn);
      getByText(/Lock Gate/);
    })
  })
})
