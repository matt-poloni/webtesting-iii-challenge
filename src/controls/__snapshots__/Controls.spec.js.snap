// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Snapshots should match Closed/Locked snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={false}
    onClick={[Function]}
  >
    Unlock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={true}
    onClick={[Function]}
  >
    Open Gate
  </button>
</div>
`;

exports[`<Controls /> Snapshots should match Closed/Unlocked snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={false}
    onClick={[Function]}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={false}
    onClick={[Function]}
  >
    Open Gate
  </button>
</div>
`;

exports[`<Controls /> Snapshots should match Open/Locked snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
    onClick={[Function]}
  >
    Unlock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={true}
    onClick={[Function]}
  >
    Close Gate
  </button>
</div>
`;

exports[`<Controls /> Snapshots should match Open/Unlocked snapshot 1`] = `
<div
  className="controls panel"
>
  <button
    className="toggle-btn"
    disabled={true}
    onClick={[Function]}
  >
    Lock Gate
  </button>
  <button
    className="toggle-btn"
    disabled={false}
    onClick={[Function]}
  >
    Close Gate
  </button>
</div>
`;
